src/3d/mesh/qgsmeshterraingenerator.cpp

```diff
@@ -171,6 +172,27 @@ void QgsMeshTerrainGenerator::readXml( const QDomElement &elem )
   mSymbol->readXml( elem.firstChildElement( "symbol" ), rwc );
 }
 
+float QgsMeshTerrainGenerator::heightAt( double x, double y, const Qgs3DMapSettings &map ) const
+{
+  Q_UNUSED( map );
+  if ( !meshLayer()->triangularMesh() )
```

Comment:
Doesn't this method need to be thread safe?

Reply:
I think this just returns pointer, so should be ok.

Reply:
yes but couldn't the user be changing the mesh layer, deleting it, etc while a background thread is calling heightAt?

Reply:
editing  is not allowed at  the moment, but you are right that maybe it is future-safe to be threadsafe here

Reply:
Well they could still delete the layer and cause a crash right now :laughing: